cluster/etcd.go

```diff
@@ -281,3 +284,35 @@ func (c *Cluster) getBackupImage() string {
 	logrus.Debugf("[etcd] Image used for etcd snapshot is: [%s]", rkeToolsImage)
 	return rkeToolsImage
 }
+
+func (c *Cluster) getRestoreImage() string {
+
+	// use etcd image for restore in case of custom system image
+	if !strings.Contains(c.SystemImages.Etcd, "rancher/mirrored-coreos-etcd") {
+		return c.SystemImages.Etcd
+	}
+
+	etcdImageTag, err := util.GetImageTagFromImage(c.SystemImages.Etcd)
+	if err != nil {
+		logrus.Errorf("[etcd] error getting restore image %v", err)
```

Comment:
Please specify the correct error for each as there is no way to know where it failed exactly (besides the `err` obviously). This can be specified better, i.e. `error getting image tag from restore image` or something, so it can be looked up more easily when encountered.

Reply:
updated.